django/core/exceptions.py

```diff
@@ -179,6 +179,36 @@ def __str__(self):
     def __repr__(self):
         return 'ValidationError(%s)' % self
 
+    def __eq__(self, other):
+        if hasattr(self, 'error_dict'):
+            if not hasattr(other, 'error_dict'):
+                return False
+            return self.error_dict == other.error_dict
+        elif hasattr(self, 'message'):
+            return (
+                hasattr(other, 'message') and
+                self.message == other.message and
+                self.code == other.code and
+                self.params == other.params
+            )
+
+        if not hasattr(other, 'error_list'):
+            return False
+        if len(self.error_list) != len(other.error_list):
+            return False
+        # error order in error_list doesn't matter for equality
+        other_error_list = other.error_list.copy()
```

Comment:
This is quite inefficient. how about this instead:
```python
for error in self.error_list:
    if not error in other.error_list:
        return False
return True
```

There's no need for making a copy of the other error list and removing items from it. Since you've already established that the two lists are of same length, if every item from one list is in the other, both lists are identical with regards to their content.

Reply:
Thanks for the fast review! That's a much better solution than what I had.